MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -809,27 +808,12 @@ private bool IsSaverEnabledForItem(IMetadataSaver saver, BaseItem item, LibraryO
         {
             var results = await provider.GetSearchResults(searchInfo, cancellationToken).ConfigureAwait(false);
 
-            var list = results.ToList();
```

Comment:
Multiple enumeration was introduced here and as a result potential bugs.
As I see, most of `GetSearchResults` implementations return arrays and there is no problem in this case.
But at least `MusicBrainzArtistProvider` returns a lazy enumeration using `yield` statement. And here we have a bug:
`MusicBrainzArtistProvider` lazily creates `RemoteSearchResult` while we iterate here to set `item.SearchProviderName` and then it creates new ones when we iterate this enumerable again outside. So we actually lose `SearchProviderName`

But I haven't tested yet